file_handling/save.py

```diff
@@ -0,0 +1,96 @@
+import os
+from mantid import mtd
+from mantid.api import IEventWorkspace
+from mantid.simpleapi import \
+    CloneWorkspace, Rebin, ConvertToDistribution, DiffractionFocussing, \
+    SaveNexusProcessed, SaveAscii, DeleteWorkspace
+
+
+def save_banks(
+        InputWorkspace,
+        Filename,
+        Title,
+        OutputDir='./',
+        Binning=None,
+        GroupingWorkspace=None):
+    """
+    Saves input workspace to processed NeXus file in specified
+    output directory with optional rebinning and grouping
+    (to coarsen) the output in a bank-by-bank manner. Mainly
+    wraps Mantid `SaveNexusProcessed` algorithm.
+
+    :param InputWorkspace: Mantid workspace to save out
+    :type InputWorkspace: MatrixWorkspace
+    :param Filename: Filename to save output
+    :type Filename: str
+    :param Title: A title to describe the saved workspace
+    :type Title: str
+    :param OutputDir: Output directory to save the processed NeXus file
+    :type OutputDir: path str
+    :param Binning: Optional rebinning of event workspace.
+                    See `Rebin` in Mantid for options
+    :type Binning: dbl list
+    :param GroupingWorkspace: A workspace with grouping
+                              information for the output spectra
+    :type GroupWorkspace: GroupWorkspace
+    """
+
+    # Make a local clone
+    CloneWorkspace(InputWorkspace=InputWorkspace, OutputWorkspace="tmp")
+    tmp_wksp = mtd["tmp"]
```

Comment:
If we name the `tmp_wskp` `__tmp` then it will not show up in the ADS to users whilst executing. Mantid has the concept of "Invisible workspaces" which are those that start with `__` in their name. These are not shown in the ADS unless the `Show invisible workspaces` setting is True in the config.